### PR TITLE
fixes #1639: run with unprivileged user and bump java version to 8.0.292.hs

### LINKED ISSUE #1639: Docker image - root required
**Describe the bug**
The Docker image runs on root by default which is both insecure and prevents the container from running in secured environments (like Gitlab CI).
If the user is specified (i.e. by providing `--user 1000` flag), the container fails.

**Sample**
```
❯ docker container run --user 1000 --rm -ti springcloud/spring-cloud-contract:3.0.2 ./gradlew clean build

ERROR: JAVA_HOME is set to an invalid directory: /root/.sdkman/candidates/java/current/

Please set the JAVA_HOME variable in your environment to match the
location of your Java installation.
```

### PR DESCRIPTION
**Pull request by bastiandoetsch**

### CODE REVIEW

File: docker/spring-cloud-contract-docker/project/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.4-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.1-bin.zip
```
Comment (marcingrzejszczak) [inline docker/spring-cloud-contract-docker/project/gradle/wrapper/gradle-wrapper.properties:3]: Let's leave the previous Gradle version

File: docker/spring-cloud-contract-docker/project/gradlew [hunk 2]
```diff
@@ -72,7 +72,7 @@ case "`uname`" in
   Darwin* )
     darwin=true
     ;;
-  MINGW* )
+  MSYS* | MINGW* )
```
Comment (marcingrzejszczak) [inline docker/spring-cloud-contract-docker/project/gradlew:75]: Same here as far as Gradle version is concerned.

**Review by marcingrzejszczak — CHANGES_REQUESTED** [review]
LGTM apart from the Gradle version change

File: docker/spring-cloud-contract-docker/project/gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (bastiandoetsch) [reply docker/spring-cloud-contract-docker/project/gradle/wrapper/gradle-wrapper.properties:3]: Wasn't even intentional, tbh. I'll change it.

**Comment by marcingrzejszczak** [comment]
Thanks @bastiandoetsch for your contribution!

**Pull request merged**